bin/pot-to-php.js

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env node
+
+const gettextParser = require( 'gettext-parser' );
+const _isEmpty = require( 'lodash/isEmpty' );
+const path = require( 'path' );
+const fs = require( 'fs' );
+
+const TAB = '\t';
+const NEWLINE = '\n';
+
+const fileHeader = [
+	'<?php',
+	'/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */',
+	'$generated_i18n_strings = array(',
+].join( NEWLINE ) + NEWLINE;
+
+const fileFooter = NEWLINE + [
+	');',
+	'/* THIS IS THE END OF THE GENERATED FILE */',
+].join( NEWLINE ) + NEWLINE;
+
+/**
+ * Escapes single quotes.
+ *
+ * @param {string} input The string to be escaped.
+ * @return {string} The escaped string.
+ */
+function escapeSingleQuotes( input ) {
+	return input.replace( /'/g, '\\\'' );
+}
+
+/**
+ * Converts a translation parsed from the POT file to lines of WP PHP.
+ *
+ * @param {Object} translation The translation to convert.
+ * @param {string} textdomain The text domain to use in the WordPress translation function call.
+ * @return {string} Lines of PHP that match the translation.
+ */
+function convertTranslationToPHP( translation, textdomain ) {
+	let php = '';
+
+	// The format of gettext-js matches the terminology in gettext itself.
+	let original = translation.msgid;
+	const comments = translation.comments;
+
+	if ( ! _isEmpty( comments ) ) {
```

Comment:
Pretty sure that while we're including the file reference in the PHP file, we're omitting the comments that would be picked up in the PHP string extraction, both text of the original comment and file reference.

https://codex.wordpress.org/I18n_for_WordPress_Developers#Descriptions

https://github.com/WordPress/wordpress-develop/blob/b15919c8dd7e4085b854b03a352db497a39f114a/tools/i18n/extract.php#L17